mempool/src/pool/orphans/mod.rs

```diff
@@ -263,6 +273,27 @@ impl TxOrphanPool {
 
         n_evicted
     }
+
+    /// Remove orphans for given originator
+    pub fn remove_by_origin(&mut self, origin: TxOrigin) -> usize {
+        let mut n_removed = 0;
+
+        while let Some(iid) = self.pick_by_origin(origin) {
+            let _ = self.remove_at(iid);
+            n_removed += 1;
+        }
+
+        n_removed
+    }
+
+    /// Pick one orphan from given origin
+    fn pick_by_origin(&self, origin: TxOrigin) -> Option<InternalId> {
+        self.maps
+            .by_origin
+            .range((origin, InternalId::ZERO)..=(origin, InternalId::MAX))
+            .map(|(_origin, iid)| *iid)
+            .next()
+    }
```

Comment:
Not sure if it's worth it or not, but if we're likely to have many transactions from peers, maybe have another level here - so that on cleanup you can just delete the whole origin in a single call rather than iterating?

Reply:
There are not going to be many transactions from one peer. The whole orphan pool is limited to 100 transactions by default so there's your upper bound. Will be even more restricted once some sort of rate limiting is implemented

Reply:
In addition to that, you still need to iterate to get it out of all the other indices